src/app/src/main/kotlin/org/imperial/mrc/hint/db/UserRepository.kt

```diff
@@ -1,50 +1,68 @@
 package org.imperial.mrc.hint.db
 
+import org.imperial.mrc.hint.emails.EmailManager
+import org.imperial.mrc.hint.emails.PasswordEmailTemplate
 import org.imperial.mrc.hint.exceptions.UserException
 import org.pac4j.core.profile.CommonProfile
 import org.pac4j.sql.profile.DbProfile
 import org.pac4j.sql.profile.service.DbProfileService
 import org.springframework.stereotype.Component
+import java.security.SecureRandom
+import java.util.*
 
-interface UserRepository
-{
-    fun addUser(email: String, password: String)
+interface UserRepository {
+    fun addUser(email: String, password: String?)
     fun removeUser(email: String)
     fun getUser(email: String): CommonProfile?
     fun updateUserPassword(user: CommonProfile, password: String)
 }
 
 @Component
-class DbProfileServiceUserRepository(private val profileService: DbProfileService): UserRepository
-{
-    override fun addUser(email: String, password: String)
-    {
-        if (getUser(email) != null)
-        {
+class DbProfileServiceUserRepository(private val profileService: DbProfileService,
+                                     private val emailManager: EmailManager) : UserRepository {
+
+    override fun addUser(email: String, password: String?) {
+        if (getUser(email) != null) {
             throw UserException("User already exists")
         }
 
         val profile = DbProfile()
         profile.build(email, mapOf("username" to email))
 
-        profileService.create(profile, password)
+        val pw = if (password.isNullOrEmpty()) {
```

Comment:
It feels a little bit weird for it to be the user repo that is responsible for sending emails - outside its jurisdiction as a repo. Would maybe feel more natural for there to be a logic class doing that - or put the logic in the cli itself?